runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java

```diff
@@ -248,6 +251,17 @@ public void syncWithRetryUnsafe(VersionLockedObject vlo,
      */
     public abstract void addTransaction(AbstractTransactionalContext tc);
 
+    /**
+     * Add an object that needs extra processing right before commit happens
+     *
+     * @param preCommitListener The context of the object that needs extra processing
+     *                         along with its lambda.
+     */
+    public abstract void addPreCommitListener(PreCommitListener preCommitListener);
```

Comment:
Shouldnt it be overridden in WriteAfterWriteTransactionalContext as well?

Reply:
WriteAfterWrite inherits from OptimisticTransaction and will end up inheriting this method too.